subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestFrameworkAutoDetectionIntegrationTest.groovy

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.integtests.fixtures.DefaultTestExecutionResult
+import org.gradle.testing.fixture.JUnitCoverage
+import org.gradle.testing.fixture.TestNGCoverage
+
+class TestFrameworkAutoDetectionIntegrationTest extends AbstractIntegrationSpec {
+
+    def setup() {
+        buildFile << """
+            plugins {
+                id("java")
+            }
+            repositories {
+                ${mavenCentralRepository()} 
+            }
+        """
+    }
+
+    def "uses JUnit Platform if junit-platform-engine is on the classpath"() {
+        given:
+        buildFile << """
+            dependencies {
+                testImplementation('org.testng:testng:${TestNGCoverage.NEWEST}')
+                testImplementation('org.junit.jupiter:junit-jupiter:${JUnitCoverage.LATEST_JUPITER_VERSION}')
+            }
+        """
+        withTestClass('JupiterTestClass', 'org.junit.jupiter.api.Test')
+
+        when:
+        succeeds("test")
+
+        then:
+        assertTestPassed('JupiterTestClass')
+    }
+
+    def "uses JUnit Platform if JUnit 4 and junit-vintage-engine are on the classpath"() {
```

Comment:
I was just poking around with auto test detection and was able to make this still test pass when using the wrong test framework (JUnit vs JUnitPlatform).

Reply:
You're right. This should also include a test for another engine or simply check for the `TestFramework` being used.